casanovo/config.py

```
@@ -162,6 +166,55 @@ def __init__(self, config_file: Optional[str] = None):
             )
 
+    def _builtin_configs(self):
+        return {
+            name.removeprefix("_").removesuffix("_config"): value
+            for name, value in vars(self).items()
+            if name.endswith("_config") and isinstance(value, Path)
+        }
+
+    def _resolve_config_file(self, config_file: Optional[str]) -> Path:
+        if config_file is None:
+            warnings.warn(
```

Ruff (0.15.15)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)